Author: Attila Nagy
Subject: Complain about unsupported Python version
Tag: idea
Tag: issue
Tag: reviewed
Message-Id: <AANLkTimb4NR3gQyNtdcH955W8xX2x0U-q4EAevjEh_qD@mail.gmail.com>
Date: Wed, 30 Jun 2010 14:49:02 +0200

Currently, if one attempts to launch Hk on a system with Python 2.4
installed, xe gets this message:

    $ ./hk.py
    Traceback (most recent call last):
      File "./hk.py", line 29, in ?
        import hklib
      File "/home/attish/heapkeeper/hklib.py", line 451
        with open(filename, 'r') as f:

This happens due to the missing ``with`` keyword on this platform.
However, seeing this message, the user might (and is likely to) think
that there is a bug in Hk, and may instantly decide to abandon it
instead of switching to a newer Python version.

We should check the version of Python very early in the initialization
process, and suggest upgrading if it is unsupported.
